File: Js/matrixDsum.js
var diagonalSum = function (mat) {
  let sum = 0
  for (let i = 0; i < mat.length; i++) {
    for (let j = 0; j < mat[i].length; j++) {
      if (i == j) {
        sum = sum + mat[i][j]
      } else if (i + j == mat[i].length - 1) {
        sum = sum + mat[i][j]
      }
    }
  }
  console.log(sum)
  return sum
}

// let mat = [
//   [1, 1, 1, 1],
//   [1, 1, 1, 1],
//   [1, 1, 1, 1],
//   [1, 1, 1, 1],
// ]
// let mat = [
//   [1, 2, 3],
//   [4, 5, 6],
//   [7, 8, 9],
// ]
let mat = [[5]]
diagonalSum(mat)

File: python/matrixDsum.py
from operator import le


class Solution:
    def diagonalSum(self, mat):
        sum=0
        for i in range(len(mat)):
            for j in range(len(mat[i])):
                if(i==j):
                    sum=sum+mat[i][j]
                elif(i+j ==len(mat)-1):
                    sum=sum+mat[i][j]
        return sum

mat = [
  [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9],]
x=Solution()
y=x.diagonalSum(mat)
print(y)
